app/support/reservations/date_support.rb

```diff
@@ -172,25 +172,29 @@ def set_all_split_times
   end
 
   def assign_reserve_from_params(params)
-    # need to be reset to nil so the individual pieces will
-    # take precedence
-    self.reserve_start_at = nil
-    self.reserve_end_at   = nil
     reserve_attrs = params.slice(:reserve_start_date, :reserve_start_hour, :reserve_start_min, :reserve_start_meridian,
                                  :duration_mins,
                                  :reserve_start_at, :reserve_end_at)
+    # need to be reset to nil so the individual pieces will
+    # take precedence, but only reset them if we're going to overwrite them
+
+    self.reserve_start_at = nil if reserve_attrs.keys.any? { |k| k.to_s.start_with?("reserve_start") }
```

Comment:
I was not a huge fan of making changes to this part of the code, but it seemed the most expedient thing to do. What was happening was since we were only submitting the `actual_end_*` fields, the reservation timestamps as well as `actual_start_at` was being cleared out at first.